### ISSUE TITLE
session lifetime hardcoded in guard.SessionWrapper

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7245]
|<img alt="douglas's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| douglas reported|
|-|-|
|Trac ID|trac#248|
|Type|defect|
|Created|2003-09-01 03:59:40Z|

Attachments:

* [guard.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ca3/ca3799b8ff860c55da009a5675031b8644cdf7e3/79354b27865b1cd0fa8ca1e6ac85736423f3b82e.diff) (1128 bytes) - added by douglas on 2003-09-01 03:59:40Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__248 248
type__defect defect
reporter__douglas douglas
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1062388780000000 1062388780000000
changetime__1467620139007989 1467620139007989
version__ 
owner__ 
cc__douglas
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="douglas's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|douglas commented|
|-|-|

```
#!html
<pre>
Using a dynademo-ish approach, there is no way of setting
the expiry time of a wrapped session.  With the attached
diff, you could go:

r = guard.SessionWrapper(....))
r.setLifetime(3600)
return r

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Fixed in CVS - SessionWrappers now have an attribute
sessionLifetime that you can set.

r = guard.SessionWrapper(....)
r.sessionLifetime = 3600

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset c1d3adf8fa60fe8caad977c2604827b5cbf4eb08
```
#!CommitTicketReference repository="" revision="c1d3adf8fa60fe8caad977c2604827b5cbf4eb08"
Merge pull request #248 from twisted/8524-rodrigc-file: Replace file() with open() in twisted/protocols

Replace file() with open() in twisted/protocols

Author: rodrigc, Lukasa
Reviewer: adiroiban
Fixes: #8548
```
